### PR TITLE
[full-ci] Add keyboard selection / navigation

### PR DESCRIPTION
**Pull request by lookacat**
## Description
See https://github.com/owncloud/web/issues/6029

Needs:
https://github.com/owncloud/owncloud-design-system/pull/2218

## Related Issue

- Fixes https://github.com/owncloud/web/issues/6029

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by JammingBen** [comment]
Issues I've found while testing:

* When navigating via arrow keys and exceeding the list (e.g. when my selection is on the last resource, and I keep pressing the down key), then the selection vanishes, but the outline still persists on the last element. When clicking up again, the resource above gets selected:

https://user-images.githubusercontent.com/50302941/175500628-e62fc778-1bbe-4f97-bc48-129746aa3272.mov

* When selecting multiple resources via "shift+mouseclick" that are **above** the first selection, the selected row (outline) does not jump to the last clicked element (I am shift-clicking on "222222.md" in the clip).

https://user-images.githubusercontent.com/50302941/175501042-a242c951-9f04-4802-a9a3-61ab13c741af.mov

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue [hunk 1]
```diff
@@ -701,7 +713,7 @@ export default defineComponent({
        * Triggered when a default action is triggered on a file
        * @property {object} resource resource for which the event is triggered
        */
-      this.$emit('fileClick', resource)
+      // this.$emit('fileClick', resource)
```
Comment (JammingBen) [inline packages/web-app-files/src/components/FilesList/ResourceTable.vue:716]: Is this not needed anymore?

File: packages/web-app-files/src/store/mutations.js [hunk 2]
```diff
@@ -91,21 +91,29 @@ export default {
   SET_CLIPBOARD_ACTION(state, action) {
     state.clipboardAction = action
   },
+  SET_LATEST_SELECTED_FILE(state, file) {
```
Comment (JammingBen) [inline packages/web-app-files/src/store/mutations.js:94]: I'd rather call this `SET_LATEST_SELECTED_FILE_ID` to make clear the method does not expect a resource object here.

```suggestion
  SET_LATEST_SELECTED_FILE_ID(state, fileId) {
```

**Review by JammingBen — DISMISSED** [review]

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone1</strong> https://drone.owncloud.com/owncloud/web/26459/49/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Review by tbsbdr — DISMISSED** [review]
uha, this adds so much usability to ownCloud web 💌

I have one thing I want to request to change:

## Scroll with selected focus:

### Current behavior:
1. upload ~100 items in one folder
2. select the first item
3. hit ⬇️  about 20 times
4. focused item is not visible in the viewport

https://user-images.githubusercontent.com/26610733/176429004-38699c62-4856-4a5a-9f8f-5a039a00aca7.mp4


### Expected behavior
1. upload ~100 items in one folder
2. select the first item
3. hit ⬇️  about 20 times
4. focused item is visible in the viewport: user always sees the current focused item

https://user-images.githubusercontent.com/26610733/176428954-19a1f0e8-9746-41a0-b0c2-a703713362d5.mp4

File: packages/web-app-files/src/store/mutations.js [hunk 3]
```diff
@@ -91,21 +91,29 @@ export default {
   SET_CLIPBOARD_ACTION(state, action) {
     state.clipboardAction = action
   },
+  SET_LATEST_SELECTED_FILE_ID(state, file) {
```
Comment (kulmann) [inline packages/web-app-files/src/store/mutations.js:101]: Please rename the `file` param to `fileId`.

File: tests/e2e/support/objects/app-files/utils/sidebar.ts [hunk 4]
```diff
@@ -51,7 +51,7 @@ export const openPanel = async ({ page, name }: { page: Page; name: string }): P
   const nextPanel = page.locator(`#sidebar-panel-${name}-item`)
 
   await Promise.all([
-    locatorUtils.waitForEvent(nextPanel, 'focus'),
+    // locatorUtils.waitForEvent(nextPanel, 'focus'),
```
Comment (kulmann) [inline tests/e2e/support/objects/app-files/utils/sidebar.ts:54]: what about this comment? does the row need to be deleted or does it need to be uncommented again?

File: packages/web-pkg/src/components/sidebar/SideBar.vue [hunk 5]
```diff
@@ -168,22 +170,13 @@ export default defineComponent({
       },
       immediate: true
     },
-    loading: {
-      handler() {
+    sidebarClosed: {
```
Comment (kulmann) [inline packages/web-pkg/src/components/sidebar/SideBar.vue:173]: erm... we only need to init the visibility observers when the sidebar is being opened, right? from my understanding the watcher should look like this:

```js
sidebarClosed: {
      handler: function (closed) {
        if (closed) {
          return
        }
        this.$nextTick(() => {
          this.initVisibilityObserver()
        })
      },
      immediate: true
    }
```

**Review by kulmann — CHANGES_REQUESTED** [review]
Needs a changelog item

File: tests/e2e/support/objects/app-files/utils/sidebar.ts (hunk 4 above)
Comment (lookacat) [reply tests/e2e/support/objects/app-files/utils/sidebar.ts:54]: no it needs to be deleted

File: packages/web-runtime/package.json [hunk 6]
```diff
@@ -24,7 +24,7 @@
     "luxon": "^2.4.0",
     "marked": "^4.0.12",
     "oidc-client": "1.11.5",
-    "owncloud-design-system": "^14.0.0-alpha.2",
+    "owncloud-design-system": "14.0.0-alpha.4",
```
Comment (pascalwengerter) [inline packages/web-runtime/package.json:27]: Please update the changelog item also;)

**Review by pascalwengerter — DISMISSED** [review]

**Review by kulmann — CHANGES_REQUESTED** [review]
Some small nitpicks, see comments. But most importantly: somehow I can only use keyboard actions at all when I mouse-click a checkbox. From that moment on the keyboard actions work. When I lose the highlight on the checkboxes keyboard actions don't work anymore.

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**